cmd/erasure-createfile.go

```diff
@@ -46,6 +46,9 @@ func erasureCreateFile(disks []StorageAPI, volume, path string, reader io.Reader
 			return nil, 0, nil, traceError(rErr)
 		}
 		if rErr == io.EOF {
+			if bytesWritten == 0 && !allowEmpty {
```

Comment:
PutObject: first part is can be empty, all other parts not
Even if higher label may prevent this (?!), I would consider this as a bug.
Found this by adding test cases for erasreCreateFile...